I'll get the CI tests fixed related to the feature change. I wound running pytest on may laptop (a mac) and not on a Linux box, so I think it skipped the test. I'm struggling to get pytest to run, though.


```
% pytest -vx tests/pytests/unit/grains/test_core.py::test_almalinux_kitten_os_grains
<frozen importlib._bootstrap>:1047: ImportWarning: NaclImporter.find_spec() not found; falling back to find_module()
=========================================================================================================== test session starts ===========================================================================================================
platform linux -- Python 3.11.7, pytest-8.1.1, pluggy-1.4.0 -- /home/jharr/workspace/github/jamesharr/salt/.venv/bin/python3.11
cachedir: .pytest_cache
max open files: soft=524288; hard=524288
salt-transport: zeromq
rootdir: /home/jharr/workspace/github/jamesharr/salt
configfile: pytest.ini
plugins: flaky-3.8.1, httpserver-1.0.8, timeout-2.3.1, helpers-namespace-2021.12.29, custom-exit-code-0.3.0, system-statistics-1.0.2, pyfakefs-5.3.1, anyio-4.1.0, subtests-0.4.0, skip-markers-1.5.0, shell-utilities-1.8.0, salt-factories-1.0.1
collected 1 item                                                                                                                                                                                                                          

tests/pytests/unit/grains/test_core.py::test_almalinux_kitten_os_grains FAILED                                                                                                                                                      [100%]

================================================================================================================ FAILURES =================================================================================================================
_____________________________________________________________________________________________________ test_almalinux_kitten_os_grains _____________________________________________________________________________________________________

    @pytest.mark.skip_unless_on_linux
    def test_almalinux_kitten_os_grains():
        """
        Test that 'os' grain will properly detect AlmaLinux Kitten
        """
        # /etc/os-release data taken from an AlmaLinux Kitten VM, install ISO
        # AlmaLinux-Kitten-10-20241018.0-x86_64_v2-minimal.iso. At the time of
        # writting, there was no docker image for Kitten
        _os_release_data = {
            "NAME": "AlmaLinux Kitten",
            "VERSION": "10 (Lion Cub)",
            "ID": "almalinux",
            "ID_LIKE": "rhel centos fedora",
            "VERSION_ID": "10",
            "PLATFORM_ID": "platform:el10",
            "PRETTY_NAME": "AlmaLinux Kitten 10 (Lion Cub)",
            "ANSI_COLOR": "0;34",
            "LOGO": "fedora-logo-icon",
            "CPE_NAME": "cpe:/o:almalinux:almalinux:10::baseos",
            "HOME_URL": "https://almalinux.org/",
            "DOCUMENTATION_URL": "https://wiki.almalinux.org/",
            "VENDOR_NAME": "AlmaLinux",
            "VENDOR_URL": "https://almalinux.org/",
            "BUG_REPORT_URL": "https://bugs.almalinux.org/",
            "ALMALINUX_MANTISBT_PROJECT": "AlmaLinux-10",
            "ALMALINUX_MANTISBT_PROJECT_VERSION": "10",
            "REDHAT_SUPPORT_PRODUCT": "AlmaLinux",
            "REDHAT_SUPPORT_PRODUCT_VERSION": "10",
        }
        expectation = {
            "os": "AlmaLinux",
            "os_family": "RedHat",
            "oscodename": "Lion Cub",
            "osfullname": "AlmaLinux",
            "osrelease": "10",
            "osrelease_info": (10, 0),
            "osmajorrelease": 10,
            "osfinger": "AlmaLinux-10",
        }
>       _run_os_grains_tests(_os_release_data, {}, expectation)

tests/pytests/unit/grains/test_core.py:1197: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tests/pytests/unit/grains/test_core.py:647: in _run_os_grains_tests
    os_grains = core.os_data()
salt/grains/core.py:2618: in os_data
    (
/usr/lib64/python3.11/platform.py:799: in __iter__
    (self.processor,)
/usr/lib64/python3.11/functools.py:1001: in __get__
    val = self.func(instance)
/usr/lib64/python3.11/platform.py:794: in processor
    return _unknown_as_blank(_Processor.get())
/usr/lib64/python3.11/platform.py:739: in get
    return func() or ''
/usr/lib64/python3.11/platform.py:762: in from_subprocess
    return subprocess.check_output(
/usr/lib64/python3.11/subprocess.py:466: in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
/usr/lib64/python3.11/subprocess.py:548: in run
    with Popen(*popenargs, **kwargs) as process:
/usr/lib64/python3.11/subprocess.py:1018: in __init__
    self.stdout = io.TextIOWrapper(self.stdout,
/usr/lib64/python3.11/encodings/__init__.py:99: in search_function
    mod = __import__('encodings.' + modname, fromlist=_import_tail,
.venv/lib64/python3.11/site-packages/mock/mock.py:1178: in __call__
    return _mock_self._mock_call(*args, **kwargs)
.venv/lib64/python3.11/site-packages/mock/mock.py:1182: in _mock_call
    return _mock_self._execute_mock_call(*args, **kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

_mock_self = <MagicMock name='__import__' id='140589785572240'>, args = ('encodings.utf_8',), kwargs = {'fromlist': ['*'], 'level': 0}, self = <MagicMock name='__import__' id='140589785572240'>
effect = <function _run_os_grains_tests.<locals>._import_mock at 0x7fdd9c3849a0>

    def _execute_mock_call(_mock_self, *args, **kwargs):
        self = _mock_self
        # separate from _increment_mock_call so that awaited functions are
        # executed separately from their call, also AsyncMock overrides this method
    
        effect = self.side_effect
        if effect is not None:
            if _is_exception(effect):
                raise effect
            elif not _callable(effect):
                result = next(effect)
                if _is_exception(result):
                    raise result
            else:
>               result = effect(*args, **kwargs)
E               TypeError: _run_os_grains_tests.<locals>._import_mock() got an unexpected keyword argument 'fromlist'

.venv/lib64/python3.11/site-packages/mock/mock.py:1245: TypeError
============================================================================================================ warnings summary =============================================================================================================
salt/utils/pycrypto.py:27
  /home/jharr/workspace/github/jamesharr/salt/salt/utils/pycrypto.py:27: DeprecationWarning: 'crypt' is deprecated and slated for removal in Python 3.13
    import crypt

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
========================================================================================================= short test summary info =========================================================================================================
FAILED tests/pytests/unit/grains/test_core.py::test_almalinux_kitten_os_grains - TypeError: _run_os_grains_tests.<locals>._import_mock() got an unexpected keyword argument 'fromlist'
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! stopping after 1 failures !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
====================================================================================================== 1 failed, 1 warning in 1.22s =======================================================================================================
```

--- a/changelog/66991.added.md
+++ b/changelog/66991.added.md
@@ -1,0 +1,1 @@
+Add detection for OS grains when running in [AlmaLinux Kitten](https://wiki.almalinux.org/release-notes/kitten-10.html)

--- a/salt/grains/core.py
+++ b/salt/grains/core.py
@@ -1775,6 +1775,7 @@ _OS_NAME_MAP = {
     "cloudlinux": "CloudLinux",
     "virtuozzo": "Virtuozzo",
     "almalinux": "AlmaLinux",
+    "almalinuxk": "AlmaLinux",
     "pidora": "Fedora",
     "scientific": "ScientificLinux",
     "synology": "Synology",
@@ -1846,6 +1847,7 @@ _OS_FAMILY_MAP = {
     "CloudLinux": "RedHat",
     "Virtuozzo": "RedHat",
     "AlmaLinux": "RedHat",
+    "AlmaLinux Kitten": "RedHat",
     "OVS": "RedHat",
     "OEL": "RedHat",
     "XCP": "RedHat",
@@ -2492,7 +2494,7 @@ def _osrelease_data(os, osfullname, osrelease):
             grains["osrelease_info"],
         )
 
-    if os in ("Debian", "FreeBSD", "OpenBSD", "NetBSD", "Mac", "Raspbian"):
+    if os in ("Debian", "FreeBSD", "OpenBSD", "NetBSD", "Mac", "Raspbian", "AlmaLinux",):
         os_name = os
     else:
         os_name = osfullname

--- a/tests/pytests/unit/grains/test_core.py
+++ b/tests/pytests/unit/grains/test_core.py
@@ -449,10 +449,10 @@ def test_gnu_slash_linux_in_os_name():
     orig_import = __import__
     built_in = "builtins"
 
-    def _import_mock(name, *args):
+    def _import_mock(name, *args, **kwargs):
         if name == "lsb_release":
             raise ImportError("No module named lsb_release")
-        return orig_import(name, *args)
+        return orig_import(name, *args, **kwargs)
 
     # - Skip the first if statement
     # - Skip the selinux/systemd stuff (not pertinent)
@@ -526,10 +526,10 @@ def test_suse_os_from_cpe_data():
     orig_import = __import__
     built_in = "builtins"
 
-    def _import_mock(name, *args):
+    def _import_mock(name, *args, **kwargs):
         if name == "lsb_release":
             raise ImportError("No module named lsb_release")
-        return orig_import(name, *args)
+        return orig_import(name, *args, **kwargs)
 
     distro_mock = MagicMock(
         return_value=("SUSE Linux Enterprise Server ", "12", "x86_64")
@@ -595,10 +595,10 @@ def _run_os_grains_tests(os_release_data, os_release_map, expectation):
     orig_import = __import__
     built_in = "builtins"
 
-    def _import_mock(name, *args):
+    def _import_mock(name, *args, **kwargs):
         if name == "lsb_release":
             raise ImportError("No module named lsb_release")
-        return orig_import(name, *args)
+        return orig_import(name, *args, **kwargs)
 
     suse_release_file = os_release_map.get("suse_release_file")
 

